### ISSUE TITLE
crashes on import due to inability to encode a filename by h5py

### ISSUE THREAD

**Issue by yarikoptic** [nipy/nipype#2497]
### Summary
Trying with basic installation in debian stretch:
```shell
root@e4d9bde617b2:/# python -c 'import nipype; print nipype'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/nipype/__init__.py", line 51, in <module>
    from .pipeline import Node, MapNode, JoinNode, Workflow
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/__init__.py", line 11, in <module>
    from .engine import Node, MapNode, JoinNode, Workflow
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/engine/__init__.py", line 12, in <module>
    from .workflows import Workflow
  File "/usr/lib/python2.7/dist-packages/nipype/pipeline/engine/workflows.py", line 36, in <module>
    from ...interfaces.base import (traits, TraitedSpec, TraitDictObject,
  File "/usr/lib/python2.7/dist-packages/nipype/interfaces/__init__.py", line 14, in <module>
    from .utility import IdentityInterface, Rename, Function, Select, Merge
  File "/usr/lib/python2.7/dist-packages/nipype/interfaces/utility/__init__.py", line 10, in <module>
    from .base import (IdentityInterface, Rename, Select, Split, Merge,
  File "/usr/lib/python2.7/dist-packages/nipype/interfaces/utility/base.py", line 19, in <module>
    import nibabel as nb
  File "/usr/lib/python2.7/dist-packages/nibabel/__init__.py", line 45, in <module>
    from .loadsave import load, save
  File "/usr/lib/python2.7/dist-packages/nibabel/loadsave.py", line 18, in <module>
    from .imageclasses import all_image_classes
  File "/usr/lib/python2.7/dist-packages/nibabel/imageclasses.py", line 16, in <module>
    from .minc2 import Minc2Image
  File "/usr/lib/python2.7/dist-packages/nibabel/minc2.py", line 31, in <module>
    h5py, have_h5py, setup_module = optional_package('h5py')
  File "/usr/lib/python2.7/dist-packages/nibabel/optpkg.py", line 98, in optional_package
    pkg = __import__(name, fromlist=fromlist)
  File "/usr/lib/python2.7/dist-packages/h5py/__init__.py", line 60, in <module>
    from .tests import run_tests
  File "/usr/lib/python2.7/dist-packages/h5py/tests/__init__.py", line 15, in <module>
    from . import old, hl
  File "/usr/lib/python2.7/dist-packages/h5py/tests/old/__init__.py", line 4, in <module>
    from . import ( test_attrs,
  File "/usr/lib/python2.7/dist-packages/h5py/tests/old/test_group.py", line 41, in <module>
    fsencode(u"α")
  File "/usr/lib/python2.7/dist-packages/h5py/_hl/compat.py", line 68, in fsencode
    return filename.encode(encoding, errors)
  File "/usr/lib/python2.7/dist-packages/future/utils/surrogateescape.py", line 59, in surrogateescape_handler
    decoded = replace_surrogate_encode(mystring)
  File "/usr/lib/python2.7/dist-packages/future/utils/surrogateescape.py", line 86, in replace_surrogate_encode
    raise exc
NameError: global name 'exc' is not defined
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib/python2.7/dist-packages/nipype/utils/config.py", line 360, in free_display
    from .. import config
SystemError: Parent module 'nipype' not loaded, cannot perform relative import
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib/python2.7/dist-packages/nipype/utils/config.py", line 360, in free_display
    from .. import config
SystemError: Parent module 'nipype' not loaded, cannot perform relative import
```
not sure if that is just a bug in future (I have 0.15.2-4), h5py (2.7.0-1), locale (tried specifying various LC_ALL) or more to it... but it is quite heavy for h5py to test filesystem encoding all the way at the import time since imho should depend on the save location etc.

So just FYI

**Comment by effigies** [comment]
This seems to be an open issue at PythonCharmers/python-future#256. And agreed that h5py should not be diving that deep into tests at import time, and this should probably be resolved by h5py/h5py#1014.

This seems to be getting pulled in through nibabel, where we could check against import-time errors besides `ImportError`. I'm not sure whether there's anything to be done on the nipype end of things, though.

**Comment by kitchell** [comment]
Is there a solution to this? I am running into the same issue

**Comment by effigies** [comment]
I've just asked h5py to post another release candidate. If they do, then you can try installing the release candidate, which should avoid this particular issue. (If you're actually using h5py functionality, there's a chance you could run into other bugs they're ironing out.)

If you don't need h5py (e.g. for MINC2), you could set up an environment where it's not available. Or we could try patching nibabel, as I suggested above. (There isn't a time table for a new nibabel release at this point, though.)

**Comment by effigies** [comment]
Should be resolved with h5py >= 2.8.1.
